web/js/panel.js

```diff
@@ -1395,10 +1824,17 @@ var $setWidgetState = function($id, $newState) {
             case "drawn" :
                 if ($widget.widgetType == "layoutturnout") {
                     $drawTurnout($widget);
+                } else if ($widget.widgetType == "layoutSlip") {
+                    $drawSlip($widget);
+                    $newState = $oldState;      // so widget state doesn't change for slips (we set it when we click)
```

Comment:
My original thought was that JMRI would communicate slip status via turnout messages (CHOSED & THROWN)… which I would finagle into the new slip state (STATE_AC, STATE_AD, STATE_BC & STATE_BD)… but actually JMRI does sends the correct slip state… and not the individual slip turnout states… so this code was originally there to suppress any unintentional slip state changes… and was unnecessarily so Line 1829 has now been deleted.